lib/filters/shoutcloud.coffee

```diff
@@ -0,0 +1,29 @@
+request = require 'request'
+
+module.exports =
+  name: "ShoutCloud"
+  description: ""
+
+  applies: (req) ->
+    req.query? and req.query.shoutcloud?
+
+  process: (req, res, message, subtitle, next) =>
+
+    str = "#{message}**seperator**#{subtitle}"
+
+    request.post {
+      headers: 'content-type': 'application/json'
+      url: 'HTTP://API.SHOUTCLOUD.IO/V1/SHOUT'
+      body: '{"INPUT": "'+str+'"}'
+    }, (error, response, body) =>
+      return @onError(req, res) if error?
+      try
+        str = JSON.parse(body).body['OUTPUT'].split('**SEPERATOR**')
+        next(req,res,str[0],str[1])
+      catch
+        return @onError(req, res)
+
+  onError: (req,res) ->
+    res.status 408
+    res.send('408 Bad Gateway - Upstream soundcloud.io is down.')
```

Comment:
1) You’re blaming a server error on the client.
2) You’re blaming an entire range of possible errors on a bad upstream gateway
3) You’re using the client socket timeout status code to reflect this entire range of possible errors.

I only comment because you seemed to be doing this purposely, rather than the worse-is-better default of just spitting out a 500 and saying ‘fuck off’.

:)